panics from tengo

Hi I am getting panics from tengo.
```
 runtime error: index out of range [129] with length 128
 /usr/local/go/src/runtime/panic.go:88 (0x430a32)
 /go/pkg/mod/github.com/d5/tengo/v2@v2.1.0/vm.go:101 (0x8b0859)
 /go/pkg/mod/github.com/d5/tengo/v2@v2.1.0/vm.go:77 (0x8aa70e)
 /go/pkg/mod/github.com/d5/tengo/v2@v2.1.0/script.go:198 (0x8a5aba)
```
Cannot pinpoint to the exact tengo code which yields this panic  (will try to isolate the tengo code which yields this)

```
fmt := import(`fmt`)

// Partition function for quicksort
partition := func(arr, low, high) {
    pivot := arr[high]
    i := low
    j := low

    for i < high {
        if arr[i] < pivot {
            temp := arr[i]
            arr[i] = arr[j]
            arr[j] = temp
            j = j + 1
        }
        i = i +1
    }
    temp := arr[j]
    arr[j] = arr[high]
    arr[high] = temp
    return j
}

// Function to call the partition function
// and perform quick sort on the array
quicksort_impl := func(arr, low, high) {
    if low < high {
        pivot := partition(arr, low, high)
        quicksort_impl(arr, low, pivot-1)
        quicksort_impl(arr, pivot + 1, high)
        return arr
    }
}

// Function to perform the insertion sort
insertion_sort_impl := func(arr, low, n) {
    fmt.println(`insertsort`)
    i := low + 1
    for i < n + 1 {
        val := arr[i]
        j := i
        for j > low && arr[j-1] > val {
                arr[j] = arr[j-1]
                j = j - 1
            arr[j]= val
        }
        i = i + 1
    }
}

// Hybrid function -> Quick + Insertion sort
hybrid_quicksort_impl := func(arr, low, high) {
    fmt.println(`hqs`, ` low `,low, ` high `,high)
    for low < high {
        fmt.println(`hqs 1`)
        // If the size of the array is less
        // than threshold apply insertion sort
        // and stop recursion

        if high - low + 1 < 10 {
            fmt.println(`hqs insert sort`)
            insertion_sort_impl(arr, low, high)
            return
        } else {
            fmt.println(`hqs 2`)
            pivot := partition(arr, low, high)


            // Optimised quicksort which works on
            // the smaller arrays first

            // If the left side of the pivot
            // is less than right, sort left part
            // and move to the right part of the array

            if pivot - low < high - pivot {
                fmt.println(`hqs 3`)
                hybrid_quicksort_impl(arr, low, pivot-1)
                low = pivot + 1
            } else {

                // If the right side of pivot is less
                // than left, sort right side and
                // move to the left side
                 fmt.println(`hqs 4`)
                hybrid_quicksort_impl(arr, pivot + 1, high)
                high = pivot-1
            }
        }
    }

    // uncomment nexy line for not having panic
    //fmt.println(`exit`)
}



// smaller arrays dont panic
vals := [1.0921152053480714, 0.9856846177133762, 0.9572347452764391, 1.1418269134461454, 0.9216375387720384, 1.065515063777526, 1.035390602066132, 1.1084141321033882, 0.9142705615917031, 0.9116642457930951, 0.2]
hybrid_quicksort_impl(vals, 0, len(vals) -1)
```

this gives me:
```
hqs low 0 high 10
hqs 1
hqs 2
hqs 3
hqs low 0 high -1
panic: runtime error: index out of range [221] with length 220

goroutine 1 [running]:
github.com/d5/tengo/v2.(*VM).run(0xc0000f4000)
    /go/src/github.com/d5/tegno/vm.go:101 +0x5f7c
github.com/d5/tengo/v2.(*VM).Run(0xc0000f4000, 0xc0000f4000, 0x400)
    /go/src/github.com/d5/tegno/vm.go:77 +0xcf
main.CompileAndRun(0xc00000e6c0, 0xc0000ba000, 0xa9e, 0xc9e, 0x7fffeed5f92d, 0x39, 0x0, 0xc000062180)
    /go/src/github.com/d5/tegno/cmd/tengo/main.go:131 +0x1e8
main.main()
    /go/src/github.com/d5/tegno/cmd/tengo/main.go:70 +0x323
```

--- a/compiler.go
+++ b/compiler.go
@@ -1192,6 +1192,7 @@ func (c *Compiler) optimizeFunc(node parser.Node) {
 	var lastOp parser.Opcode
 	var appendReturn bool
 	endPos := len(c.scopes[c.scopeIndex].Instructions)
+	newEndPost := len(newInsts)
 	iterateInstructions(newInsts,
 		func(pos int, opcode parser.Opcode, operands []int) bool {
 			switch opcode {
@@ -1204,6 +1205,8 @@ func (c *Compiler) optimizeFunc(node parser.Node) {
 				} else if endPos == operands[0] {
 					// there's a jump instruction that jumps to the end of
 					// function compiler should append "return".
+					copy(newInsts[pos:],
+						MakeInstruction(opcode, newEndPost))
 					appendReturn = true
 				} else {
 					panic(fmt.Errorf("invalid jump position: %d", newDst))

